dist/sizzle.js

```diff
@@ -465,26 +465,51 @@ function createButtonPseudo( type ) {
  * @param {Boolean} disabled true for :disabled; false for :enabled
  */
 function createDisabledPseudo( disabled ) {
-	// Known :disabled false positives:
-	// IE: *[disabled]:not(button, input, select, textarea, optgroup, option, menuitem, fieldset)
-	// not IE: fieldset[disabled] > legend:nth-of-type(n+2) :can-disable
+
+	// Known :disabled false positives: fieldset[disabled] > legend:nth-of-type(n+2) :can-disable
 	return function( elem ) {
 
-		// Check form elements and option elements for explicit disabling
-		return "label" in elem && elem.disabled === disabled ||
-			"form" in elem && elem.disabled === disabled ||
+		// Only certain elements can match :enabled or :disabled
+		// https://html.spec.whatwg.org/multipage/scripting.html#selector-enabled
+		// https://html.spec.whatwg.org/multipage/scripting.html#selector-disabled
+		if ( "form" in elem ) {
+
+			// Check for inherited disabledness on relevant non-disabled elements:
+			// * listed form-associated elements in a disabled fieldset
+			//   https://html.spec.whatwg.org/multipage/forms.html#category-listed
+			//   https://html.spec.whatwg.org/multipage/forms.html#concept-fe-disabled
+			// * option elements in a disabled optgroup
+			//   https://html.spec.whatwg.org/multipage/forms.html#concept-option-disabled
+			// All such elements have a "form" property.
+			if ( elem.parentNode && elem.disabled === false ) {
+				return "label" in elem ?
```

Comment:
> Go has named return values

Well, `return` and ternary not always go together

> Rust has if-expressions

Relevant - https://github.com/rust-lang/rust/issues/1698#issuecomment-3705066

> As it turns out, this function is now in a form where Uglify can produce the exact nested ternaries that I want, so I've implemented the statement-based version.

> I'm blocked on handling of expression parentheses

We can disable parentheses rules at this point I think (keep in mind that eslint has a lot of issues so team "buried" by them, so there is a good chance those are not gonna be resolved in the near future) important part is start to lint maybe not fully, but at least partially. Like doing the work you started in https://github.com/jquery/sizzle/pull/330

> Like @dmethvin said, the consequences weren't obvious.

If ticket is closed it doesn't mean we can't restart discussion in there or we can re-open it